msal/build.gradle

```diff
@@ -124,6 +134,38 @@ task sourcesJar(type: Jar) {
     destinationDir = reporting.file("$project.buildDir/outputs/jar/")
 }
 
+task createMsalCoverageReportTask() {
+    group = "Msal Reporting"
+    description = "Generate Coverage Report tasks for every build Flavour"
+    android.libraryVariants.all { variant ->
+        def buildVariant = variant.name
+        def testTaskName = "test${buildVariant.capitalize()}UnitTest"
+
+        // Tasks for generating Coverage reports for every buildVariant
+        task "${buildVariant}MsalCoverageReport" (type:JacocoReport, dependsOn: ["$testTaskName", "create${buildVariant.capitalize()}CoverageReport"]) {
+            group = "Msal Reporting"
+            description = "Generate Msal Coverage Reports(UnitTest & AndroidTests) on the ${buildVariant.capitalize()}"
+
+            def coverageSourceDirs = "${projectDir}/src/main/java"
+            def fileFilter = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/Manifest*.*', '**/*Test.*']
+            def javaClasses = fileTree(
+                    dir: "${buildDir}/intermediates/javac/${buildVariant}/classes",
+                    excludes: fileFilter
+            )
+            sourceDirectories.from = files([coverageSourceDirs])
+            additionalSourceDirs.from = files(coverageSourceDirs)
```

Comment:
What are additionalSourceDirs?

Reply:
According to documentation,
sourceDirectories - Source sets that coverage should be reported for.
additionalSourceDirs - Additional source dirs for the classes coverage data is being reported for,
So after investigation, sourceDirectories is only sufficient to cover all the classess for the coverage report, additionalSourceDirs is redundant here